ddtrace/contrib/celery/utils.py

```diff
@@ -33,27 +34,32 @@
 )
 
 
-def set_tags_from_context(span, context):
-    # type: (Span, Dict[str, Any]) -> None
-    """Helper to extract meta values from a Celery Context"""
+def exclude_context_filter(key, value):
+    # Skip this key if it is not set
+    if value is None or value == "":
+        return True
 
-    for key, tag_name in TAG_KEYS:
-        value = context.get(key)
+    # Skip `timelimit` if it is not set (its default/unset value is a
+    # tuple or a list of `None` values
+    if key == "timelimit" and all(_ is None for _ in value):
+        return True
 
-        # Skip this key if it is not set
-        if value is None or value == "":
-            continue
+    # Skip `retries` if its value is `0`
+    if key == "retries" and value == 0:
+        return True
 
-        # Skip `timelimit` if it is not set (its default/unset value is a
-        # tuple or a list of `None` values
-        if key == "timelimit" and all(_ is None for _ in value):
-            continue
+    return False
 
-        # Skip `retries` if its value is `0`
-        if key == "retries" and value == 0:
-            continue
 
-        span.set_tag(tag_name, value)
+def set_tags_from_context(span, context):
+    # type: (Span, Dict[str, Any]) -> None
+    """Helper to extract meta values from a Celery Context"""
+
+    context_tags = [
+        (tag_name, context.get(key)) for key, tag_name in TAG_KEYS if not exclude_context_filter(key, context.get(key))
+    ]
```

Comment:
I think this would be better written as not a comprehension.

```python
// I don't have a strong opinion on this name, is "context_value" appropriate?
def should_skip_context_value(key, value):
    ...
    return False


def set_tags_from_context(span, context):
    context_tags = []
    for key, tag_name in TAG_KEYS:
        value = context.get(key)
        if should_skip_context_value(key, value):
            continue

        context_tags.append((tag_name, value))

    set_flattened_tags(span, context_tags)
```


I find this easier to read, and will avoid the double `context.get(key)` calls.